code/modules/projectiles/projectile/hookshot.dm

```diff
@@ -244,3 +244,99 @@
 		sharpness = 2
 	..(A)
 
+//Wind-up Boxes///////////////////////////////////////////////////////////////////
+
+/obj/item/projectile/hookshot/whip/bootbox
+	name = "boot-in-a-box"
+	icon_state = "spring"
+	icon_name = "spring"
+	nodamage = 0
+	damage = 0
+	sharpness = 0
+	kill_count = 20 //range is defined by maxlength so this prevents animation issues
+	failure_message = ""
+	can_tether = FALSE
+	var/windUp = 0
+	var/springForce = 0
+
+/obj/item/projectile/hookshot/whip/bootbox/OnFired()
+	..()
+	var/obj/item/weapon/gun/hookshot/whip/bootbox/T = shot_from
+	if(istype(T))
+		src.windUp = T.windUp
+		src.springForce = T.springForce //inherits all the oomph from the box itself
+		damage = (windUp + springForce*5)
+		T.maxlength += springForce
```

Comment:
well this does explain you resetting maxlength above, but it does mean that you can't really varedit these objects to go farther, this doesn't seem like the best implmenetation

Reply:
Do you mean you'd prefer it doesn't increase range with springForce or just that I should make it more flexible?